rdbms/src/main/java/org/spine3/server/storage/jdbc/util/Closeables.java

```diff
@@ -27,20 +27,41 @@
  *
  * @author Dmytro Dashenkov.
  */
-@SuppressWarnings("UtilityClass")
 public class Closeables {
 
     private Closeables() {
     }
 
+    /**
+     * Closes all of the passed {@linkplain AutoCloseable AutoCloseables}.
+     *
+     * <p>In case of an {@linkplain Exception} on the {@link AutoCloseable#close()} tries to close
+     * all the rest instances.
+     *
+     * <p>The very first {@linkplain Exception} caught will be propagated in an {@link IllegalStateException}.
+     *
+     * @param closeables instances to close
+     * @throws IllegalStateException if {@linkplain AutoCloseable#close() close()} throws an {@link Exception}
+     */
     public static void closeAll(Iterable<? extends AutoCloseable> closeables) {
         checkNotNull(closeables);
-        try {
-            for (AutoCloseable closable : closeables) {
+        Exception exception = null;
+        AutoCloseable faultyInstance = null;
+        for (AutoCloseable closable : closeables) {
+            try {
                 closable.close();
+            } catch (Exception e) {
+                if (exception == null) {
```

Comment:
This way we remember only the last one. But we need to remember all. Otherwise we lose information on all previous errors.

We need a custom exception, say `MultipleCloseException`, which would accept `Iterable<Exception>` to constructor. We gather all the exceptions, then create `MultipleCloseException` with them, and pass the as the cause to `IllegalStateException`.

Please pay attention to `toString()` of `MultipleCloseException`. It should print all the enclosed exceptions nicely.